src/trix/models/composition.coffee

```diff
@@ -69,52 +70,8 @@ class Trix.Composition extends Trix.BasicObject
     @setSelection(endPosition)
     @notifyDelegateOfInsertionAtRange([startPosition, endPosition])
 
-  breakFormattedBlock: ->
-    position = @getPosition()
-    range = [position - 1, position]
-
-    document = @document
-    {index, offset} = document.locationFromPosition(position)
-    block = document.getBlockAtIndex(index)
-
-    if block.getBlockBreakPosition() is offset
-      document = document.removeTextAtRange(range)
-      range = [position, position]
-    else
-      if block.text.getStringAtRange([offset, offset + 1]) is "\n"
-        range = [position - 1, position + 1]
-      else if offset - 1 isnt 0
-        position += 1
-
-    newDocument = new Trix.Document [block.removeLastAttribute().copyWithoutText()]
-    @setDocument(document.insertDocumentAtRange(newDocument, range))
-    @setSelection(position)
-
   insertLineBreak: ->
-    [startPosition, endPosition] = @getSelectedRange()
-    startLocation = @document.locationFromPosition(startPosition)
-    endLocation = @document.locationFromPosition(endPosition)
-    block = @document.getBlockAtIndex(endLocation.index)
-
-    if block.hasAttributes()
-      if block.isListItem()
-        if block.isEmpty()
-          @decreaseListLevel()
-          @setSelection(startPosition)
-        else if startLocation.offset is 0
-          document = new Trix.Document [block.copyWithoutText()]
-          @insertDocument(document)
-        else
-          @insertBlockBreak()
-      else
-        if block.isEmpty()
-          @removeLastBlockAttribute()
-        else if block.text.getStringAtRange([endLocation.offset - 1, endLocation.offset]) is "\n"
-          @breakFormattedBlock()
-        else
-          @insertString("\n")
-    else
-      @insertString("\n")
+    Trix.LineBreakInsertion.perform(this)
```

Comment:
It’s great that we were able to reduce all the duplication in the `should*` methods by pulling them into a new object, but I think ultimately the change went too far and is a net negative for this method. 

We’ve outsourced its entire implementation to another file. None of the other Composition methods are written this way. Compare it to [the previous version](https://github.com/basecamp/trix/blob/c9ea6c794914a8bfec9ac8bd188a5f9a1e6349a7/src/trix/models/composition.coffee#L97-L115) which has a very clear set of rules spelled out in full context of the composition.

What do you think about moving the insertion responsibilities back into Composition, and treating this new model as something you query via the `should*` methods?